projects/bird-atlas/src/app/+grid/grid-index/grid-index-map/grid-index-map.component.ts

```diff
@@ -0,0 +1,174 @@
+import {
+  AfterViewInit, ChangeDetectionStrategy, Component, ElementRef, EventEmitter, Input,
+  NgZone, OnChanges, OnDestroy, Output, SimpleChanges, ViewChild
+} from '@angular/core';
+import { LajiMap, DataOptions, TileLayersOptions, Lang } from 'laji-map';
+import { environment } from 'projects/bird-atlas/src/env/environment';
+import { convertYkjToGeoJsonFeature } from 'projects/laji/src/app/root/coordinate-utils';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { filter, takeUntil } from 'rxjs/operators';
+import { AtlasGrid, AtlasGridSquare } from '../../../core/atlas-api.service';
+
+type ColorMode = 'activityCategory' | 'speciesCount';
+
+interface MapData {
+  grid: AtlasGrid;
+  data: DataOptions;
+};
+
+// Performance optimization: precomputing gradients
+// import { discreteColorGradient } from './color-math';
+// const acGradient = discreteColorGradient('FAFAD1', '4B57A4', 6);
+// const scGradient = discreteColorGradient('FAFAD1', '4B57A4', 4);
+const acGradient = ['fafad1', 'cbf2ad', '8ee59a', '72d5b9', '59a3c2', '4b57a4'];
+const scGradient = ['fafad1', '9bea98', '69cfc6', '4b57a4'];
+
+const getAtlasActivityCategoryColor = (ac: string): string => (acGradient[ac]);
```

Comment:
I'm little confused about this change... Wasn't `ac` supposed to be something like `MY.atlasActivityCategoryEnum0` so how can you use it to access array? I can see that you have changed the format of the test data that is generated in this component but how about the production data?

Reply:
Yep, accidentally changed the format of the test data and then compensated by changing the logic. Fixed.